pkgs/by-name/ma/material-maker/package.nix

```
@@ -0,0 +1,144 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  autoPatchelfHook,
+  copyDesktopItems,
+  makeDesktopItem,
+  godot3-export-templates,
+  godot3-headless,
+  alsa-lib,
+  libGL,
+  libGLU,
+  libpulseaudio,
+  libX11,
+  libXcursor,
+  libXext,
+  libXfixes,
+  libXi,
+  libXinerama,
+  libXrandr,
+  libXrender,
+  zlib,
+  udev,
+}:
+stdenv.mkDerivation rec {
+  pname = "material-maker";
+  version = "1.3";
+
+  src =
+    if stdenv.isLinux then
+      fetchurl {
+        url = "https://github.com/RodZill4/material-maker/releases/download/1.3/material_maker_1_3_linux.tar.gz";
+        hash = "sha256-Y8+ZwXy3zqnsxqqaZeVgFSxLzmUkq+rBzbq8tEDc8/g=";
+      }
+    else
+      throw "unsupported platform";
+
+  icon = fetchurl {
+    url = "https://github.com/RodZill4/material-maker/raw/1.3/icon.png";
+    hash = "sha256-cu4aD8WbeYuISeub0Mt7Z/2e4/9BZF+QwsqYFo82Ge8=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    copyDesktopItems
+    godot3-headless
+  ];
+
+  buildInputs = [
+    alsa-lib
+    libGL
+    libGLU
+    libX11
+    libXcursor
+    libXext
+    libXfixes
+    libXi
+    libXinerama
+    libXrandr
+    libXrender
+    zlib
+    udev
+  ];
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "material-maker";
+      exec = "material-maker";
+      icon = "material-maker";
+      desktopName = "Material Maker";
+      comment = meta.description;
+      categories = [
+        "Graphics"
+        "Development"
+      ];
+      keywords = [ "material" ];
+    })
+  ];
```

We can use some information from the upstream [desktop file](https://github.com/RodZill4/material-maker/blob/0268009d02061c566c281a0d65adde67f2f1caa1/material_maker/misc/linux/io.github.RodZill4.Material-Maker.desktop).

Also, we don't have to call `makeDesktopItem` from a list as we only have one desktop file.

```suggestion
  desktopItems = makeDesktopItem {
    name = "material-maker";
    exec = "material-maker";
    icon = "material-maker";
    genericName = "Procedural texture generation and 3D model painting tool";
    desktopName = "Material Maker";
    comment = meta.description;
    terminal = false;
    categories = [
      "Graphics"
      "3DGraphics"
    ];
    keywords = [ "material" ];
  };
```